crates/next-core/src/manifest.rs

```diff
@@ -177,3 +182,26 @@ impl ContentSource for DevManifestContentSource {
         ))
     }
 }
+
+/// PageSortKey is necessary because the next.js client code looks for matches
+/// in the order the pages are sent in the manifest,if they're sorted
+/// alphabetically this means \[slug] and \[\[catchall]] routes are prioritized
+/// over fixed paths, so we have to override the ordering with this.
+#[derive(Ord, PartialOrd, Eq, PartialEq)]
+enum PageSortKey {
+    Static(String),
+    Slug,
+    CatchAll,
+}
+
+impl From<&str> for PageSortKey {
```

Comment:
I think we can bound the lifetime of `PageSortKey` by the `&str`, so we don't need to clone into a `String`?
```suggestion
impl<'a> From<&'a str> for PageSortKey<'a> {
```

Reply:
I was assuming the `&str` wouldn't survive after returning out of the `sort_by_cached_key` lambda function, but let me check

Reply:
yeah

```rs
error: lifetime may not live long enough
  --> crates\next-core\src\manifest.rs:77:39
   |
77 |         routes.sort_by_cached_key(|s| s.split('/').map(PageSortKey::from).collect::<Vec<_>>());
   |                                    -- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ returning this value requires that `'1` must outlive `'2`
   |                                    ||
   |                                    |return type of closure is Vec<PageSortKey<'2>>
   |                                    has type `&'1 std::string::String`
```